rtslib/root.py

```diff
@@ -327,30 +336,35 @@ def err_func(err_str):
             if 'name' not in so:
                 err_func("'name' not defined in storage object %d" % index)
                 continue
-            try:
-                so_cls = so_mapping[so['plugin']]
-            except KeyError:
-                err_func("'plugin' not defined or invalid in storageobject %s" % so['name'])
-                continue
-            kwargs = so.copy()
-            dict_remove(kwargs, ('exists', 'attributes', 'plugin', 'buffered_mode', 'alua_tpgs'))
-            try:
-                so_obj = so_cls(**kwargs)
-            except Exception as e:
-                err_func("Could not create StorageObject %s: %s" % (so['name'], e))
-                continue
 
-            # Custom err func to include block name
-            def so_err_func(x):
-                return err_func("Storage Object %s/%s: %s" % (so['plugin'], so['name'], x))
-
-            set_attributes(so_obj, so.get('attributes', {}), so_err_func)
-
-            for alua_tpg in so.get('alua_tpgs', {}):
-               try:
-                   ALUATargetPortGroup.setup(so_obj, alua_tpg, err_func)
-               except RTSLibALUANotSupported:
-                   pass
+            if (not storage_object and not target) or (storage_object and so['name'] == storage_object):
+                try:
+                    so_cls = so_mapping[so['plugin']]
+                except KeyError:
+                    err_func("'plugin' not defined or invalid in storageobject %s" % so['name'])
+                    continue
```

Comment:
BTW, how about here when there has failures ? 

Since we have already found it but it fails to continue, IMO there is no need to continue the loop too.

But since the failure branch is rare to happen, I am okay with it and it will up to you.

Thanks.

Reply:
@lxbsz yeah, I thought it was a rare case, so left it there :-)

But yeah, since you have pointed it, it won't hurt addressing it. Will do.

And in python there is no goto :) which would have mad life easy.